src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/NamedTypeSymbolReferenceFinder.cs

```diff
@@ -45,22 +51,48 @@ protected override bool CanFind(INamedTypeSymbol symbol)
 
         protected override async Task<ImmutableArray<Document>> DetermineDocumentsToSearchAsync(
             INamedTypeSymbol symbol,
+            HashSet<string>? globalAliases,
             Project project,
             IImmutableSet<Document>? documents,
             FindReferencesSearchOptions options,
             CancellationToken cancellationToken)
+        {
+            using var _ = ArrayBuilder<Document>.GetInstance(out var result);
+
+            await AddDocumentsToSearchAsync(symbol.Name, project, documents, result, cancellationToken).ConfigureAwait(false);
+            if (globalAliases != null)
+            {
+                foreach (var alias in globalAliases)
+                    await AddDocumentsToSearchAsync(alias, project, documents, result, cancellationToken).ConfigureAwait(false);
+            }
+
+            result.AddRange(await FindDocumentsAsync(
+                project, documents, symbol.SpecialType.ToPredefinedType(), cancellationToken).ConfigureAwait(false));
+
+            result.AddRange(await FindDocumentsWithGlobalAttributesAsync(
+                project, documents, cancellationToken).ConfigureAwait(false));
+
+            return result.ToImmutable();
+        }
+
+        private static async Task AddDocumentsToSearchAsync(
+            string throughName,
```

Comment:
Whats a "through" name?

Reply:
Will doc. But say you have a type 'Foo', you could reference it through that name, or through a global alias name if you had 'global alias Bar = Foo'